src/routers/user/membership.ts

```diff
@@ -0,0 +1,65 @@
+import express, { Request, Response, NextFunction } from 'express';
+import validate from '../../middleware/validate';
+import routeValidator from './routeValidator';
+import User from '../../models/user';
+import { MEMBERSHIP_STATUS } from '../../types/user';
+import createHttpError from 'http-errors';
+import validator from 'validator';
+
+const router = express.Router();
+
+// update user's membership status
+router.patch(
+  '/',
+  routeValidator('/membership'),
+  validate,
+  async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const user = await User.findById(req.query.id);
+      if (!user) {
+        throw createHttpError(404, 'Cannot find user with that id');
+      }
+      const membershipStatus = req.body.membershipStatus;
+      if (!Object.values(MEMBERSHIP_STATUS).includes(membershipStatus)) {
+        throw createHttpError(422, 'invalid membership status');
+      }
```

Comment:
We can make this into a custom validator and let `express-validator` handle this

Reply:
Sure!